api/tests/routes/pro/patch_collective_offer_test.py

```diff
@@ -379,6 +380,74 @@ def when_user_is_not_attached_to_offerer(self, app, client):
         ]
         assert CollectiveOffer.query.get(offer.id).name == "Old name"
 
+    def test_cannot_update_offer_with_used_booking(self, client):
+        # Given
+        offer = CollectiveOfferFactory()
+        UsedCollectiveBookingFactory(
+            collectiveStock__collectiveOffer=offer,
+        )
+        offerers_factories.UserOffererFactory(
+            user__email="user@example.com",
+            offerer=offer.venue.managingOfferer,
+        )
+
+        # When
+        data = {
+            "name": "New name",
+        }
+        response = client.with_session_auth("user@example.com").patch(
+            f"/collective/offers/{humanize(offer.id)}", json=data
+        )
+
+        # Then
+        assert response.status_code == 403
+        assert response.json == {"offer": "Les offres utilisées ou remboursées ne sont pas modifiable."}
+
+    def test_patch_collective_offer_replacing_by_non_unknown_venue(self, client):
```

Comment:
Une variante des fameux "known unknowns" de Rumsfeld ? ;)

```suggestion
    def test_patch_collective_offer_replacing_by_unknown_venue(self, client):
```